xarray/core/dataarray.py

```diff
@@ -816,6 +816,118 @@ def swap_dims(self, dims_dict):
         ds = self._to_temp_dataset().swap_dims(dims_dict)
         return self._from_temp_dataset(ds)
 
+    def set_index(self, indexers=None, append=False, inplace=False,
+                  **kw_indexers):
+        """Set DataArray (multi-)indexes using one or more existing coordinates.
+
+        Parameters
+        ----------
+        indexers : dict, optional
```

Comment:
Another downside of using a single argument is that it is less consistent with other parts of the xarray's API which use keyword arguments (e.g., `reindex`, `sel`, etc.). IMHO the little additional verbosity is worth the gain in explicit/self-documenting here.

That said, a single `dim_levels` or `levels` argument may be more consistent with the signature you suggest below for reset_index: `reset_index(dim, levels=None,...)`. So finally I don't really know what is better.